src/app/api.ts

```diff
@@ -213,6 +218,34 @@ export interface API {
     signal?: AbortSignal
   ): Promise<SecretResponse[]>;
 
+  createSecretProjectGroups(
+    username: string,
+    secretname: string,
+    keyPairs: object,
+    signal?: AbortSignal
+  ): Promise<SecretResponse>;
+
+  createSecretProjects(
+    username: string,
+    secretname: string,
+    projectName: string,
+    keyPairs: object,
+    signal?: AbortSignal
+  ): Promise<SecretResponse>;
+
+  deleteSecretProjectGroups(
+    secretName: string,
+    username: string,
+    signal?: AbortSignal
+  ): Promise<void>;
+
+  deleteSecretProject(
+    secretName: string,
+    username: string,
+    projectName: string,
+    signal?: AbortSignal
+  ): Promise<void>;
+
```

Comment:
Why isn't there an updateSecret* function? Isn't there also a ui for updating secrets?

Reply:
Providing features for updating secrets requires an extra endpoint for retrieving the secrets key values.
This endpoint is not available for security reasons. 
So the operation of updating is equivalent to deleting and inserting again all the values.

Reply:
The backend provides a update secret endpoint. Of course you have to re-provide all the keys and values but this is a single op instead of a delete and create.

The ui, on an existing secret should also provide an update button and an update route. The create and update secret routes should be the same ('createsecret.vue' perhaps renamed to createupdatesecret.vue) and the submit operation will be a create or an update based on the route (use a property based on the route).

Reply:
I don't see the above comment as resolved.

Reply:
Probably this the above comment wasn't clear.